src/tests/utilities/push-pull-async-iterable-iterator.ts

```diff
@@ -0,0 +1,57 @@
+export class PushPullAsyncIterableIterator<T>
+  implements AsyncIterableIterator<T> {
+  public pushQueue: T[] = [];
+  private pullQueue: ((value: IteratorResult<T>) => void)[] = [];
+  private isRunning = true;
```

Comment:
This AsyncIterable will add pending items to the push queue and pending next call promises to the pull queue. New values can be added by calling `push`. This allows us more control over what is actually happening where PubSub is a black box we have no control over. It prevents us from using hacks like `setTimeout` in order to fix the timings.